### DISCUSSION TITLE
Data Importer 404 Error

### DISCUSSION THREAD

**Discussion by SpookedByRoaches** [firefly-iii/firefly-iii#7617] [Firefly III]
I've been trying to install the data-importer server on a self hosted server. I think I've configured it correctly where the server configuration is similar to the configuration for the normal ff3 app (which works great). However I keep getting a 404 error when it redirects me to the url http://importerLocation/public/token. 

I've installed the correct PHP version, configured apache to user PHP correctly (similar to the normal app), put the correct vanity_url. I've double checked all those parameters. I don't have the importer app behind a proxy, but I still made it trust proxies and that did nothing. I've reinstalled the package from composer a couple times. I've `chown`d the files in the directory to the http user. 

The storage/logs/data-import.log says
```
[2023-06-09 15:21:30] local.DEBUG: Now in App\Http\Controllers\IndexController::index
[2023-06-09 15:21:30] local.DEBUG: App\Services\Shared\Authentication\SecretManager::hasValidSecrets
[2023-06-09 15:21:30] local.DEBUG: No valid secrets, redirect to token.index
```

I can't think of anything else that I can do to fix this at the moment, what else should I do?

**Comment by PierfrancescoElia** [comment]
Hi, I am in the same situation too. Difference: I've set as directory in the virtual host the public, so I see:
<img width="345" alt="image" src="https://github.com/firefly-iii/firefly-iii/assets/8597934/60f6a2c4-c71f-46d1-b5f1-cf0282c41daa">

**Comment by JC5** [comment]
What happens when you press [Reauthenticate]?

**Comment by SpookedByRoaches** [comment]
I can't see a reauthenticate button on the normal ff3 site. And the data importer immediately takes me to the 404 page when I type the URL, so no reauthenticate button there either. I tried to put in reauthenticate as a route in the URL on both sites and they both give me a 404 error.

**Reply by JC5** [reply]
Try browsing to /flush on the data importer.

**Reply by SpookedByRoaches** [reply]
It gives me a 404 error.

**Reply by PierfrancescoElia** [reply]
It behaves the same with me too.

**Reply by JC5** [reply]
OK, so you have to make sure the /public/ part is not in the URL. Make sure you configure both so that /public is the web root.

**Reply by PierfrancescoElia** [reply]
Hi, thanks for your feedback, I can confirm my public root is the /public folder in data-importer project as you can see in the above screenshot. Is there any _php artisan something_ command that I should execute on my terminal? I basically followed this guide https://docs.firefly-iii.org/data-importer/installation/self-hosted/ , so the composer command + the .env file check, then I've created a Virtual Host in my apache configuration so that main FireFly is listening on port 8888 and importer is listening on port 8887. Thanks in advance :)

**Reply by SpookedByRoaches** [reply]
@PierfrancescoElia The web root is configured by the httpd configuration. Personally made it so that it is a virtual host. 
@JC5 I just changed the web root to the /public directory and it still didn't change.
```
<VirtualHost *:80>
    DocumentRoot "/srv/http/data-importer/public"
    ErrorLog "/var/log/httpd/data-importer-error.log"
    CustomLog "/var/log/httpd/data-importer-access.log" common
        <Directory /srv/http/data-importer/public>
                AllowOverride All
                Require all granted
        </Directory>
</VirtualHost>

```

Should anything else change there?

**Reply by PierfrancescoElia** [reply]
In my _httpd.conf_ I've added:
- Listen 8887 (so that I "enabled" another port for listening)
- Include extra/httpd-vhosts.conf

In _httpd_vhosts.conf_ I wrote:
```
# Virtual Hosts
#
# Required modules: mod_log_config

# If you want to maintain multiple domains/hostnames on your
# machine you can setup VirtualHost containers for them. Most configurations
# use only name-based virtual hosts so the server doesn't need to worry about
# IP addresses. This is indicated by the asterisks in the directives below.
#
# Please see the documentation at
# <URL:http://httpd.apache.org/docs/2.4/vhosts/>
# for further details before you try to setup virtual hosts.
#
# You may use the command line option '-S' to verify your virtual host
# configuration.

#
# Use name-based virtual hosting.
#
NameVirtualHost *:8887

#
# VirtualHost example:
# Almost any Apache directive may go into a VirtualHost container.
# The first VirtualHost section is used for all requests that do not
# match a ServerName or ServerAlias in any <VirtualHost> block.
#
<VirtualHost *:8887>
    DocumentRoot "/Applications/MAMP/htdocs/firefly-iii/data-importer/public"
    ErrorLog "logs/dummy-host.example.com-error_log"
    CustomLog "logs/dummy-host.example.com-access_log" common
</VirtualHost>
```

**Comment by PierfrancescoElia** [comment]
May the absence of Nordigen/Spectre token be the cause? I'm still waiting for Salt Edge test request.

**Reply by SpookedByRoaches** [reply]
While I don't have either services, I don't think this is the case. The importer can import from a CSV file without a dependency on Nordigen.

**Reply by JC5** [reply]
No that doesn't matter.

**Comment by MW-SAND** [comment]
This issue was fixed for me by setting AllowOverride to All.
